docs/content/contributing/coding.md

```diff
@@ -0,0 +1,89 @@
+---
+description: >
+    Coding guidelines for kcp projects.
+---
+
+# Coding Guidelines & Conventions
+
+- Always be clear about what clients or client configs target. Never use an unqualified `client`. Instead, always qualify. For example:
+    - `rootClient`
+    - `orgClient`
+    - `pclusterClient`
+    - `rootKcpClient`
+    - `orgKubeClient`
+- Configs intended for `NewForConfig` (i.e. today often called "admin workspace config") should uniformly be called `clusterConfig`
+    - Note: with org workspaces, `kcp` will no longer default clients to the "root" ("admin") logical cluster
+    - Note 2: sometimes we use clients for same purpose, but this can be harder to read
+- Cluster-aware clients should follow similar naming conventions:
+    - `crdClusterClient`
+    - `kcpClusterClient`
+    - `kubeClusterClient`
+- `clusterName` is a kcp term. It is **NOT** a name of a physical cluster. If we mean the latter, use `pclusterName` or similar.
+- Qualify "namespace"s in code that handle up- and downstream, e.g. `upstreamNamespace`, `downstreamNamespace`, and also `upstreamObj`, `downstreamObj`.
+- Logging:
+  - Use the `fmt.Sprintf("%s|%s/%s", clusterName, namespace, name` syntax.
+  - Default log-level is 2.
+  - Controllers should generally log (a) **one** line (not more) non-error progress per item with `klog.V(2)` (b) actions like create/update/delete via `klog.V(3)` and (c) skipped actions, i.e. what was not done for reasons via `klog.V(4)`.
+- When orgs land: `clusterName` or `fooClusterName` is always the fully qualified value that you can stick into obj.ObjectMeta.ClusterName. It's not necessarily the `(Cluster)Workspace.Name` from the object. For the latter, use `workspaceName` or `orgName`.
```

Comment:
`When orgs land:` - what did you mean here? 

`you can stick into obj.ObjectMeta.ClusterName` -> `you can use as obj.ObjectMeta.ClusterName value`